### ISSUE TITLE
[HOLD for payment 2025-01-02] [Medium] Create Trip Details

### ISSUE THREAD

**Issue by stitesExpensify** [Expensify/App#37823]
## VIP-Travel Implementation Issue
Tracking Issue: https://github.com/Expensify/Expensify/issues/333471
Project: https://github.com/orgs/Expensify/projects/112/
Relevant Design Doc Section: https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.lgcmthyxnpmx

This issue is for creating a new trip details modal:
<img width="811" alt="2024-03-06_08-26-27" src="https://github.com/Expensify/App/assets/42391420/842a70be-fbda-4577-8736-1392c10db995">

A new page, TravelDetails.ts will be created in src/pages/Travel: 
- It will receive the transactionID as a route parameter, which we’ll use to retrieve the full transaction via withOnyx, and, from there, we’ll display the travel-related information
- It will include a [HeaderWithBackButton](https://github.com/Expensify/App/blob/main/src/components/HeaderWithBackButton/index.tsx) component with either “Flight Details” or “Hotel Details” as the title, depending on whether it is a flight or a hotel booking. Similarly, it will either display a plane or a hotel icon, which we will pass to HeaderWithBackButton through the new parameter added [here](https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.i2ys1y7nerlg). - - For other travel-related expenses (Trains, car rental), the title will default to the generic “Travel Details” and the plane icon (we can change this down the line).
- The itinerary and confirmation number elements will be displayed using the existing [MenuItemWithTopDescription](https://github.com/Expensify/App/blob/main/src/components/MenuItemWithTopDescription.tsx) component with the interactive parameter set to false. For the itinerary’s arrow for flights, we can use the &#x279C; unicode character.
- A new component, TripDetailsMenuItem will be created to display the detailed itinerary (with the airline, airports, etc). It will receive the travel-related information from the transaction as a parameter and will be formatted as displayed above. Since the user can’t interact with the element in any way, there is no need for further considerations.
- The trip support link will be displayed using the existing [MenuItem](https://github.com/Expensify/App/blob/main/src/components/MenuItem.tsx) component, passing our [phone icon](https://github.com/Expensify/App/blob/d5eba656251269b238797817df3cf84800b0039a/src/components/Icon/Expensicons.ts#L102) as the icon. On click, it will take the user to Spotnana’s support page.

Routes table:
https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.gn6jbl2udlnx

<details><summary>Issue Owner</summary>Current Issue Owner: @twisterdotcom</details>

**Comment by cdOut** [comment]
Hello, I’m Tymoteusz from Software Mansion, the expert agency, and I’d like to work on this task.

**Comment by twisterdotcom** [comment]
Adding a C+ here for Travel reviews: @shubham1206agra

**Comment by twisterdotcom** [comment]
Are we waiting on anything internal here?

**Comment by cdOut** [comment]
No, Im working on creating a draft PR for this issue by tomorrow.

**Comment by twisterdotcom** [comment]
Nice, draft PR is in!

**Comment by twisterdotcom** [comment]
Same Q, answer in one place is fine: https://github.com/Expensify/App/issues/37824#issuecomment-2034635911

**Comment by cdOut** [comment]
@twisterdotcom both PRs are linked, we are currently waiting for the backend implementation, since right now we basically work on dummy data.

From what I know it has been decided that we will hold off on these PRs with further reviews until we get backend access, you can refer to [this decision here](https://github.com/Expensify/App/issues/37826#issuecomment-2023443206).

**Comment by twisterdotcom** [comment]
Still waiting on webhooks.

**Comment by twisterdotcom** [comment]
Waiting on the backend.

**Comment by twisterdotcom** [comment]
Waiting on the backend still

**Comment by twisterdotcom** [comment]
Spotnana thread for backend updates here: https://expensify.slack.com/archives/C066REXJ9D3/p1713220956933639

**Comment by twisterdotcom** [comment]
Chatting about getting the data here: https://expensify.slack.com/archives/C05S5EV2JTX/p1713784714491629?thread_ts=1713545844.230119&cid=C05S5EV2JTX

**Comment by twisterdotcom** [comment]
Update: https://expensify.slack.com/archives/C05S5EV2JTX/p1714085516706139?thread_ts=1713825695.223409&cid=C05S5EV2JTX

**Comment by twisterdotcom** [comment]
Update here: https://expensify.slack.com/archives/C05S5EV2JTX/p1714431147318139. Waiting on Spotnana.

**Comment by twisterdotcom** [comment]
@rushatgabhane any linked draft here yet?

**Comment by cdOut** [comment]
@twisterdotcom I'm already back from OOO and I have the most context on this one. Could I take this issue back since it seems like @rushatgabhane is already occupied by other projects?

**Comment by rushatgabhane** [comment]
@cdOut yes please go ahead! Thank you 🙇‍♂️

**Comment by twisterdotcom** [comment]
@cdOut I believe you should be able to get going on this now.

**Comment by cdOut** [comment]
I've replied [in this thread](https://github.com/Expensify/App/issues/37824#issuecomment-2112219586).

**Comment by twisterdotcom** [comment]
Not overdue now. On hold until we can deploy.

**Comment by twisterdotcom** [comment]
Hopefully we'll deploy tomorrow/weds.

**Comment by twisterdotcom** [comment]
This is off hold now! You should be able to get moving on it now.

**Comment by cdOut** [comment]
Thank you for the info! I’ve already picked it back up yesterday in the evening and will continue finishing it up today.

**Comment by twisterdotcom** [comment]
Is there a PR going yet here?

**Comment by cdOut** [comment]
I wasn't able to finish it last time due to the bank holidays, the PR for both this and Trip Receipt will be up by the end of Tuesday.

**Comment by cdOut** [comment]
As mentioned in [this thread on slack](https://swmansion.slack.com/archives/C05S5EV2JTX/p1717456376694299), I'm currently fixing merge conflicts and unifying certain component implementations after [Trip Room Summary](https://github.com/Expensify/App/pull/41659) has been merged.

**Comment by twisterdotcom** [comment]
EOW is the update in Slack!

**Comment by twisterdotcom** [comment]
Could we get an update in here/Slack before the end of the week please @cdOut?

**Comment by twisterdotcom** [comment]
https://expensify.slack.com/archives/C05S5EV2JTX/p1718412514464099?thread_ts=1718382488.093209&cid=C05S5EV2JTX

**Comment by twisterdotcom** [comment]
Bumped in Slack for an update today.

**Comment by twisterdotcom** [comment]
Will bump in Slack for an update later when I do the Weekly update.

**Comment by twisterdotcom** [comment]
Bumped in Slack for an update.

**Comment by twisterdotcom** [comment]
Draft PR above.

**Comment by kevinksullivan** [comment]
Demoting to HIGH, as I don't think this prevents launching to 5k customers or announcing. Also, conversation on design is ongoing here, so this may take a few more days to nail down.

https://expensify.slack.com/archives/C05S5EV2JTX/p1721737498827189

**Comment by blazejkustra** [comment]
Hey! I'm going to take over from @cdOut. Feel free to tag me when you're ready to revisit this issue 😄

**Comment by stitesExpensify** [comment]
@blazejkustra We're ready!

**Comment by blazejkustra** [comment]
Update: I started implementing this screen but I'm a bit blocked, waiting on responses on slack [here](https://expensify.slack.com/archives/C05S5EV2JTX/p1732092608910189?thread_ts=1731962682.485409&cid=C05S5EV2JTX)

**Comment by blazejkustra** [comment]
Update: PR is ready for internal review. Sorry for now updating here directly, I was doing it daily on `#vip-travel` slack channel

**Comment by blazejkustra** [comment]
We can close this issue, as PR was merged last week.

**Comment by twisterdotcom** [comment]
Payment Summary:

 @blazejkustra **does not** require payment (Contractor)
- @shubham1206agra requires payment (Needs manual offer from BZ)
